### PR TITLE
Add Stripe webhook signature verification

### PR DESCRIPTION
**Pull request by cjdenio**
Fixes #2308! Tested, too.

Y'all will just need to grab the signing secret from https://dashboard.stripe.com/webhooks and put it in the credentials file:

```diff
stripe:
  live:
    publishable_key: deer
    secret_key: emoji
+   webhook_signing_secret: dot
  test:
    publishable_key: dino
    secret_key: dot
+   webhook_signing_secret: icu
```

---

See Figure 1 for a comprehensive overview of the changes introduced in this pull request.

```mermaid
graph TD
  A([Incoming request]) --> B[Check signature]
  B -- Signature is invalid/missing? --> C[Send 400]
  B -- Signature is valid? --> D[Process webhook] --> E["Profit #128184;"]

  style C fill:red
  style D fill:green
```
<h5 align="center">Figure 1 - Comprehensive, and entirely necessary, overview</h5>

### CODE REVIEW

**Comment by maxwofford** [comment]
Thank you for adding this! Could this run conditionally if a given stripe account has a verification token? I'd like to keep it easy to run Stripe in development.

**Comment by cjdenio** [comment]
You got it! The new logic will skip signature verification if a signing secret wasn't provided (as long as it's not running in production).

**Comment by maxwofford** [comment]
Oh, to clarify, I do want to run this in development too– but it should skip if my verification tokens aren't set.

**Comment by cjdenio** [comment]
Sorry for not being clear; it'll run in development if you set a verification token.

File: app/services/stripe_service.rb [hunk 1]
```diff
@@ -20,6 +20,21 @@ def self.secret_key
     Rails.application.credentials.stripe[self.mode][:secret_key]
   end
 
+  def self.webhook_signing_secret
+    Rails.application.credentials.stripe[self.mode][:webhook_signing_secret]
+  end
+
+  def self.construct_webhook_event(payload, sig_header)
+    # Don't check signatures if we're not in production (unless there was a signing secret provided)
+    if !Rails.env.production? && self.webhook_signing_secret.blank?
```
Comment (maxwofford) [inline app/services/stripe_service.rb]:
```suggestion
    if self.webhook_signing_secret.blank?
```

**Comment by maxwofford** [comment]
Cool if I merge this as is & we add the verification tokens in a separate PR?

**Comment by cjdenio** [comment]
sounds good!

**Pull request merged**